pkg/noderesourcetopology/cache/discardreserved.go

```diff
@@ -46,12 +46,14 @@ type DiscardReserved struct {
 	rMutex         sync.RWMutex
 	reservationMap map[string]map[types.UID]bool // Key is NodeName, value is Pod UID : reserved status
 	client         ctrlclient.Client
+	lh             logr.Logger
 }
 
-func NewDiscardReserved(client ctrlclient.Client) Interface {
+func NewDiscardReserved(lh logr.Logger, client ctrlclient.Client) Interface {
 	return &DiscardReserved{
 		client:         client,
 		reservationMap: make(map[string]map[types.UID]bool),
+		lh:             lh,
```

Comment:
I noticed the logger is being used often. maybe it's worth embedding it on all the structs to save some key strokes?

Reply:
I have a strong bias towards composition rather than embedding, but if all the reviewers feel differently I'm open to reconsider